### ISSUE TITLE
Move settings from globals to the screens they control.

### ISSUE THREAD

**Issue by teryhill** [LibreHealthIO/LibreEHR#1008]
Look at the flowboard. 

![pfb](https://user-images.githubusercontent.com/4655366/36323519-2a9f0006-131f-11e8-9a15-f2a2c089857b.PNG)

The setting indicator will allow the setting fro this screen to be set.

Do the same for other screens.

**Comment by aethelwulffe** [comment]
Good idea...based on good acl controls, this helps a lot!

**Comment by sandipbhuyan** [comment]
I want to complete this task

**Comment by teryhill** [comment]
@sandipbhuyan I will assign this to you . @aethelwulffe  can you invite him

**Comment by teryhill** [comment]
@sandipbhuyan where do you plan to start, which screen?

**Comment by sandipbhuyan** [comment]
interface/main/messages/messages.php
then new/new_comprehensive.php

**Comment by teryhill** [comment]
Tell me what the settings are for them.

**Comment by sandipbhuyan** [comment]
I think this settings selector will show the content of a patient that a user want to see.
And I think I need to apply this in other pages according to that the page content will be changed.

**Comment by teryhill** [comment]
I am sorry I did not explain this well enough.

This screen contains settings that tell the system how to work

![image](https://user-images.githubusercontent.com/4655366/36326744-3048296e-132a-11e8-8179-14a7c86b0041.png)

The settings under calendar are for the calendar and flowboard.

**Comment by sandipbhuyan** [comment]
yup the settings of a page will decide the content of a page is it correct?

**Comment by teryhill** [comment]
Yes but there are no settings under globals for the two sections you mentioned.

The settings will be chosen by the user and stored for the user so the software works like they want. It is right now done under the Miscellaneous then Preferences. Look at the setting on the flowboard and then check the user settings.

**Comment by sandipbhuyan** [comment]
Sure I ll try to add settings in one page and after that i ll make a PR for it. Them I ll proceed for other pages.

**Comment by Anmol-Middha** [comment]
do we need to move the global settings to specific screens or create new settings for each screen ?

**Comment by aethelwulffe** [comment]
Better make sure that the feature isn't toggled from globals to begin with.  It would be bad to not be able to turn the feature on and off! :)

**Comment by teryhill** [comment]
create access to the settings on the screen that uses them. The globals will still be used to set the default settings. The settings on the screens are for the specific user.

**Comment by Anmol-Middha** [comment]
ok, I would like to work on this issue.

**Comment by Anmol-Middha** [comment]
Currently, I am planning to create a sort checkbox for message and reminder screen , So that a bunch of reminders can be sorted according to priority , due date or any other element which the user selects for sorting of reminders and messages  

Something like this -:
![screenshot 211](https://user-images.githubusercontent.com/26340886/36442302-50c89344-169b-11e8-9a08-19de93c94e34.png)

**Comment by apooravc** [comment]
@teryhill As I understand, a screen's settings have to be introduced to each screen from its Globals like in case of flowboard. And these settings would be user specific and Globals would be default?

**Comment by aethelwulffe** [comment]
I would build a function in a standard include...probably globals.php itself, that you can pass arguments to and pull in and display the settings control include file.  I think it would be user pref settings vs. default, but there may be exceptions to this (such as with superusers)

**Comment by aethelwulffe** [comment]
How are we progressing on this as an issue or development concept?  I think we are probably doing pretty good, but the globals interface is still growing.
  I commented somewhere that on both my local machine as well as my servers, updating globals takes a very very long time.  Like, go get a cup of coffee level.  @teryhill responded that his only takes 7 seconds.
  I wonder if this has to do with logging levels.  In terms of warnings, my server logs about fifty million or so warnings or notices when updating globals.

**Comment by apooravc** [comment]
@aethelwulffe About the updating globals part, on my machine it took like 1.5 minutes and I think it might be because the code checks if a value in field is same as in DB and if it isn't, it updates but it loops over all fields to check that. Moving globals to their respective screens might fix this.

**Comment by naveen17797** [comment]
Gettig globals to screen they control is also a good thing, but i have some other idea @apooravc @aethelwulffe , currently the global system validates and save or update every global field, if we post only the fields which are changed by means of jquery to the save part, we can easily reduce the time.

**Comment by aethelwulffe** [comment]
It really should not check each field like that, or it should build the array and push the whole thing as an array.  I hesitate to say, but I am almost convinced that globals should be crammed into a json string.  That makes looking them up in the database more difficult, but make a lot of other things easier.  A minute and a half is way, way too long.  I suspect that the time is actually related to writing to logs, and not actually just updating the table, but when we get to a couple of hundred global values, that results in a lot of individual queries.  A worst, it should do a "REPLACE INTO()VALUES();"

**Comment by aethelwulffe** [comment]
Part of the issue is the logging of lots of PHP notices too.  That should not be overlooked.
